transpiler/transpiler.go

```diff
@@ -0,0 +1,565 @@
+			return err
+		}
+		out.Writef(st, "}")
+		if st.Else != nil {
+			bs, ok := st.Else.(*ast.BlockStmt)
+			if !ok {
+				return out.Errorf(st.Else, "unsupported else statement")
+			}
+			out.Writef(st, " else {\n")
+			if err := handleBlockStmt(out, bs); err != nil {
+				return err
+			}
+			out.Writef(st, "}")
+		}
+		out.Writef(st, "\n")
+	case *ast.ReturnStmt:
+		out.Writef(st, "return ")
+		for i, r := range st.Results {
+			if i != 0 {
+				// TODO(maruel): Effectively support multiple return values.
+				out.Writef(r, ", ")
+			}
+			if err := handleExpr(out, r); err != nil {
+				return err
+			}
+		}
+		out.Writef(st, ";\n")
+	default:
+		return out.Errorf(s, "unsupported statement")
+	}
+	return nil
+}
+
+// handleBlockStmt handles a series of statements in a block delimited with "{"
+// and "}".
+func handleBlockStmt(out *output, bs *ast.BlockStmt) error {
+	for _, s := range bs.List {
+		if err := handleStmt(out, s); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// handleCallExpr handles a function call.
+func handleCallExpr(out *output, c *ast.CallExpr) error {
+	args := []string{}
+	buf := &bytes.Buffer{}
+	tmp := &output{buf, out.content, out.lines, c, nil, nil}
+	for _, a := range c.Args {
+		buf.Reset()
+		if err := handleExpr(tmp, a); err != nil {
+			return err
+		}
+		args = append(args, buf.String())
+		out.lastNode = a
+	}
+	ident, _, err := exprTypeToType(out, c.Fun)
+	if err != nil {
+		return err
+	}
+	out.Writef(c, "%s(%s)", ident, strings.Join(args, ", "))
+	return nil
+}
+
+// handleBinaryExpr handles an expression for the form "X <op> Y".
+func handleBinaryExpr(out *output, be *ast.BinaryExpr) error {
+	if err := handleExpr(out, be.X); err != nil {
+		return err
+	}
+	out.Writef(be, "%s", be.Op)
+	if err := handleExpr(out, be.Y); err != nil {
+		return err
+	}
+	return nil
+}
+
+// handleExpr handles a generic expression, like "a()", "a + b", "a != nil",
+// "a++", etc.
+func handleExpr(out *output, e ast.Expr) error {
+	switch expr := e.(type) {
+	case *ast.BasicLit:
+		// a constant
+		out.Writef(expr, "%s", expr.Value)
+	case *ast.BinaryExpr:
+		return handleBinaryExpr(out, expr)
+	case *ast.CallExpr:
+		return handleCallExpr(out, expr)
+	case *ast.Ident:
+		// identifier
+		out.Writef(expr, "%s", expr.Name)
+	case *ast.SelectorExpr:
+		// can be either a symbol from a package or a member or method dereference.
+		if err := handleExpr(out, expr.X); err != nil {
+			return err
+		}
+		// TODO(maruel): have to be converted to "->" for pointer dereference.
+		out.Writef(expr, ".")
+		return handleExpr(out, expr.Sel)
+	case *ast.StarExpr:
+		out.Writef(expr, "*")
+		return handleExpr(out, expr.X)
+	case *ast.UnaryExpr:
+		// handles an expression with only one operator, e.g. "!", "++", etc
+		out.Writef(expr, "%s", expr.Op)
+		return handleExpr(out, expr.X)
+	default:
+		return out.Errorf(e, "unsupported expr")
+	}
+	return nil
+}
+
+// typeFromExpr extracts the type from a constant.
```

Comment:
it's because it's the only moment for now when the type is needed to create a variable, as in C the type is needed, unlike for example when doing A==B where B is a constant.